packages/BRC721/src/brc721.ts

```diff
@@ -1,47 +1,26 @@
-import { NFT } from './nft'
-
-interface IBRC721 {
-  mint(to: string, name?: string, symbol?: string): Promise<NFT>
-  balanceOf(publicKey: string): Promise<number>
-  ownerOf(tokenId: string): Promise<string[]>
-  transfer(to: string, tokenId: string)
+export interface IBRC721 {
+  transfer(to: string)
 }
 
-export class BRC721 implements IBRC721 {
-  computer: any
-  masterNFT: NFT
-
-  constructor(computer: any) {
-    this.computer = computer
-  }
-
-  async mint(to: string, name?: string, symbol?: string): Promise<NFT> {
-    if (!this.masterNFT && name && symbol) {
-      this.masterNFT = this.computer.new(NFT, [to, name, symbol])
-      return this.masterNFT
-    }
-    if (this.masterNFT.name !== name || this.masterNFT.symbol !== symbol) {
-      throw new Error('Name or symbol mismatch when minting token.')
-    }
-    return this.masterNFT.mint(to)
-  }
+export class BRC721 {
+  name: string
+  symbol: string
+  _owners: string[]
+  readonly _id: string
+  readonly _rev: string
+  readonly _root: string
 
-  async balanceOf(publicKey: string): Promise<number> {
-    const revs = await this.computer.queryRevs({ publicKey, contract: NFT })
-    const objects: NFT[] = await Promise.all(revs.map((rev) => this.computer.sync(rev)))
-    objects.flatMap((object) => (object._root === this.masterNFT._root ? [object] : []))
-    return objects.length
+  constructor(to: string, name: string, symbol: string) {
+    this._owners = [to]
+    this.name = name
+    this.symbol = symbol
```

Comment:
To be able to use computer.query and have it return all smart objects of this class the constructor should have no parameters. To set the parameters we should have a separate function ``init`` that sets these parameters.